ducktape/tests/loader.py

```diff
@@ -33,25 +33,31 @@ class LoaderException(Exception):
 class TestLoader(object):
     """Class used to discover and load tests."""
 
-    def __init__(self, session_context, test_parameters=None):
+    def __init__(self, session_context, logger, injected_args=None, cluster=None):
         self.session_context = session_context
+        self.cluster = cluster
+        assert logger is not None
+        self.logger = logger
         self.test_file_pattern = DEFAULT_TEST_FILE_PATTERN
         self.test_function_pattern = DEFAULT_TEST_FUNCTION_PATTERN
 
         # A non-None value here means the loader will override the injected_args
         # in any discovered test, whether or not it is parametrized
-        self.test_parameters = test_parameters
-
-    @property
-    def logger(self):
-        return self.session_context.logger
+        self.injected_args = injected_args
 
     def parse_discovery_symbol(self, discovery_symbol):
-        """Parse command-line argument into a tuple (directory, module.py, cls_name, function_name).
+        """Parse a single 'discovery symbol'
+
+        :return tuple of form (directory, module.py, cls_name, function_name)
 
         :raise LoaderException if it can't be parsed
+
+        Examples:
+            "path/to/directory" -> ("/absolute/path/to/directory", "", "", "")
```

Comment:
Is the `/absolute/path/to/directory` part of this accurate anymore? You don't have an `abspath` call and `os.path.dirname` does not guarantee an absolute path.